src/components/carousel/__spec__.js

```diff
@@ -17,23 +17,128 @@ describe('Carousel', () => {
   });
 
   describe('componentWillMount', () => {
+    describe('when slideIndex is passed', () => {
+      it('sets the intial slide to the prop', () => {
+        const wrapper = shallow(
+          <Carousel data-element='bar' data-role='baz' slideIndex={ 1 }>
```

Comment:
These `data-element` and `data-role` should be removed. They will be applied to the component when we use them and will naturally be rendered in the tests as a result.

There are more cases of this dotted throughout the PR.